### PR TITLE
Fixed copy-locale for homepage

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes sulu/sulu-standard#780
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR fixes copy-locale on homepages.

### CODE REVIEW

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 1]
```diff
@@ -477,19 +478,23 @@ public function copyLanguage(
             $resourceLocatorStrategy = $this->resourceLocatorStrategyPool->getStrategyByWebspaceKey($webspaceKey);
         }
 
+        // The parent-document can be unknown e.g. if the requested page was the homepage
+        $parentUuid = !$parentDocument instanceof UnknownDocument ? $this->inspector->getUuid($parentDocument) : null;
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Wouldn't it be better to check if the `$document` is an instance of `HomeDocument`? Then we would not have to use `getParent` at all and save some time.

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 2]
```diff
@@ -477,19 +478,23 @@ public function copyLanguage(
...
+
         foreach ($destLocales as $destLocale) {
             $destDocument = $this->documentManager->find(
                 $uuid,
                 $destLocale
             );
             $destDocument->setLocale($destLocale);
+            $destDocument->setTitle($document->getTitle());
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: What error was caused by that line missing?

**Comment by danrot** [comment]
And please rebase 😃

File: src/Sulu/Component/Content/Mapper/ContentMapper.php (hunk 2 above)
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: the title will not be updated in the bind method.

Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: So that wasn't working correct before?

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: correct (: at least my written test failed before this line

**Comment by wachterjohannes** [comment]
@danrot fixed

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 3]
```diff
@@ -472,24 +473,30 @@ public function copyLanguage(
         }
 
         $document = $this->documentManager->find($uuid, $srcLocale);
-        $parentDocument = $this->inspector->getParent($document);
         if ($document instanceof ResourceSegmentBehavior) {
             $resourceLocatorStrategy = $this->resourceLocatorStrategyPool->getStrategyByWebspaceKey($webspaceKey);
         }
 
+        $parentUuid = null;
+        if (!$document instanceof ParentBehavior) {
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Isn't the `!` misplaced? I think this should only happen if it is of the type `ParentBehavior`...

**Pull request merged**